vm/src/main/scala/pravda/vm/DataOperations.scala

```diff
@@ -0,0 +1,701 @@
+        case BigInt(rhs) => lhs > rhs
+        case Number(rhs) => lhs > rhs
+        case _ => throw VmErrorException(WrongType)
+      }
+      case Uint16(lhs) => b match {
+        case Int8(rhs) => lhs > rhs
+        case Int16(rhs) => lhs > rhs
+        case Int32(rhs) => lhs > rhs
+        case Uint8(rhs) => lhs > rhs
+        case Uint16(rhs) => lhs > rhs
+        case Uint32(rhs) => lhs > rhs
+        case BigInt(rhs) => lhs > rhs
+        case Number(rhs) => lhs > rhs
+        case _ => throw VmErrorException(WrongType)
+      }
+      case Uint32(lhs) => b match {
+        case Int8(rhs) => lhs > rhs
+        case Int16(rhs) => lhs > rhs
+        case Int32(rhs) => lhs > rhs
+        case Uint8(rhs) => lhs > rhs
+        case Uint16(rhs) => lhs > rhs
+        case Uint32(rhs) => lhs > rhs
+        case BigInt(rhs) => lhs > rhs
+        case Number(rhs) => lhs > rhs
+        case _ => throw VmErrorException(WrongType)
+      }
+      case Number(lhs) => b match {
+        case Int8(rhs) => lhs > rhs
+        case Int16(rhs) => lhs > rhs
+        case Int32(rhs) => lhs > rhs
+        case Uint8(rhs) => lhs > rhs
+        case Uint16(rhs) => lhs > rhs
+        case Uint32(rhs) => lhs > rhs
+        case BigInt(rhs) => lhs > BigDecimal(rhs)
+        case Number(rhs) => lhs > rhs
+        case _ => throw VmErrorException(WrongType)
+      }
+      case BigInt(lhs) => b match {
+        case Int8(rhs) => lhs > rhs.toInt
+        case Int16(rhs) => lhs > rhs.toInt
+        case Int32(rhs) => lhs > rhs
+        case Uint8(rhs) => lhs > rhs
+        case Uint16(rhs) => lhs > rhs
+        case Uint32(rhs) => lhs > rhs
+        case BigInt(rhs) => lhs > rhs
+        case Number(rhs) => BigDecimal(lhs) > rhs
+        case _ => throw VmErrorException(WrongType)
+      }
+      case _ => throw VmErrorException(WrongType)
+    }
+    Bool(result)
+  }
+
+  def add(a: Data, b: Data): Data = a match {
+    case Int32(lhs) => b match {
+      case Int8(rhs) => Int32(lhs + rhs)
+      case Int16(rhs) => Int32(lhs + rhs)
+      case Int32(rhs) => Int32(lhs + rhs)
+      case Uint8(rhs) => Int32(lhs + rhs)
+      case Uint16(rhs) => Int32(lhs + rhs)
+      case Uint32(rhs) => Int32((lhs + rhs).toInt)
+      case BigInt(rhs) => Int32((lhs + rhs).toInt)
+      case Number(rhs) => Int32((lhs + rhs).toInt)
+      case _ => throw VmErrorException(WrongType)
+    }
+    case Int16(lhs) => b match {
+      case Int8(rhs) => Int16((lhs + rhs).toShort)
+      case Int16(rhs) => Int16((lhs + rhs).toShort)
+      case Int32(rhs) => Int16((lhs + rhs).toShort)
+      case Uint8(rhs) => Int16((lhs + rhs).toShort)
+      case Uint16(rhs) => Int16((lhs + rhs).toShort)
+      case Uint32(rhs) => Int16((lhs + rhs).toShort)
+      case BigInt(rhs) => Int16((lhs.toInt + rhs).toShort)
+      case Number(rhs) => Int16((lhs + rhs).toShort)
+      case _ => throw VmErrorException(WrongType)
+    }
+    case Int8(lhs) => b match {
+      case Int8(rhs) => Int32(lhs + rhs)
```

Comment:
But here it gets wider somehow.

Reply:
It was changed on my local machine. Now all operations save `lhs` type.